Parse/Platform/Files/FileState.cs

```diff
@@ -1,30 +1,48 @@
 using System;
 
-namespace Parse.Platform.Files
+namespace Parse.Platform.Files;
+
+public class FileState
 {
-    public class FileState
-    {
-        static string SecureHyperTextTransferScheme { get; } = "https";
+    static string SecureHyperTextTransferScheme { get; } = "https";
 
-        public string Name { get; set; }
+    public string Name { get; set; }
 
-        public string MediaType { get; set; }
+    public string MediaType { get; set; }
 
-        public Uri Location { get; set; }
+    public Uri Location { get; set; }
 
-        public Uri SecureLocation => Location switch
+    /// <summary>
+    /// Converts the file's location to a secure HTTPS location if applicable.
+    /// </summary>
+    public Uri SecureLocation
+    {
+        get
         {
-#warning Investigate if the first branch of this swhich expression should be removed or an explicit failure case when not testing.
+            if (Location == null)
+                throw new InvalidOperationException("Location is not set.");
 
-            { Host: "files.parsetfss.com" } location => new UriBuilder(location)
-            {
-                Scheme = SecureHyperTextTransferScheme,
+            return IsParseHostedFile(Location) ? GetSecureUri(Location) : Location;
+        }
+    }
 
-                // This makes URIBuilder assign the default port for the URL scheme.
+    /// <summary>
+    /// Checks if the file is hosted on a supported Parse file server.
+    /// </summary>
+    private static bool IsParseHostedFile(Uri location)
+    {
+        return location.Host.EndsWith("parsetfss.com", StringComparison.OrdinalIgnoreCase);
```

Comment:
I haven't found this version of the file in on my end.
Checking again

Reply:
A fair bit of the diff seems to be converting to file scoped namespaces - the diff says 183 files/15k lines changed but I don't think there was actually that much edited, right?

Reply:
Correct!

I did this change Mid port as a reflex since I usually use file scoped and by the time I noticed, I was too far gone to revert my changes, so I moved on as such.
Reason why I felt better when UT passed, At least they ensured the code is functionally upright👍🏾

Reply:
No worries. Just a bit hard to review all the files 😃 

Could you please let me know which files introduce the new functionality / have been actually changed and i'll focus my review there

Reply:
Sure thing.
I'll do my best to point them out.
let me push this commit and brb